CleanWeather/Helpers/SharedPresenter.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  SharedPresenter.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 18/11/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import UIKit
+
+class SharedPresenter {
+    
+    weak var controller: CityListPresentable?
```

Comment:
jak mamy sharedpresenter i on ma controllera
to musielibysmy tutaj zrobic tez sharedpresentable ktory to nam zalatwi

Reply:
albo w zasadzie sam uiviewcontroller tez da rade, ale lepiej to owrapowac protokołem

Reply:
Kurcze, utknąłem z tym. To tylko pokazuje, że nie do końca to rozumiem cały czas. Rozumem, że w SharedPresenter nie możemy deklarować `weak var controller: CityListPresentable?` bo to ma być reużywalne. Użyłem więc `weak var controller: UIViewController?`, ale on nie posiada metod z SharedViewController, więc użyłem `weak var controller: SharedViewController?`, ale on nie posiada methody z CityListPresentable, czyli `func displayCity(_ citiesWeather: [CityWeatherDisplayable])` Kompletnie się w tym pogubiłem. 
Czy w SharedPresenter mam zadeklarować 'weak var controller: SharedViewController?' a potem już w CityListPresenter jakoś zrobić coś (nie wiem co), żeby controller miał dostęp do methody z CityListPresentable?

Reply:
Poddaję się. Siedziałem dziś wiele godzin nad tym i nie mam zielonego pojęcia jak to zrobić, żeby w SharedPresenter mieć dwie metody dispayError i showSpinner, które w momencie kiedy będziemy dziedziczyć z SharedPresentera będą miały dostęp do method z controllera podpiętego w CityListPresenterze. Jakiś kosmos to jest dla mnie :(